Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -94,6 +94,15 @@ bool DecodeBool(pb_istream_t* stream) {
   }
 }
 
+void EncodeLong(pb_ostream_t* stream, int64_t integer_value) {
```

Comment:
Sorry I haven't noticed this before but three different names for this thing in a single line finally caught my eye :-).

These aren't "longs" and that name is misleading on 32-bit platforms (e.g. iPhone 5C which we still support) and on win64.

I see two paths forward:
  * Call them "Integer" like the Firestore proto does (you can see this in `google_firestore_v1beta1_Value_integer_value_tag`)
  * Call them Int64 like the actual type of the value

I slightly prefer the former.

Reply:
If you merge that PR down here I think this one is ready to go.